dali/test/python/test_dali_tf_dataset_eager.py

```diff
@@ -325,6 +331,20 @@ def test_tf_experimental_inputs_disabled():
                         input_datasets={"test" : tf.data.Dataset.from_tensors(np.int32([42, 42]))})
 
 
+# Test if the ValueError is raised for external source with source.
+@raises(ValueError)
+def test_tf_experimental_source_disabled():
+    pipe = Pipeline(10, 4, 0)
+    with pipe:
+        input = fn.external_source(source=lambda : np.full((4, 4), 0), batch=False)
```

Comment:
I don't like this behavior to start with - when we use a `source` (and not feeding the data to a named input from a TF dataset), then the layout should be set here, too. I just hope that it's not _cleared_ if DALIDataset doesn't provide one...?